pkgs/development/python-modules/protobuf.nix

```diff
@@ -8,6 +8,9 @@ buildPythonPackage rec {
   inherit (protobuf) name src;
   inherit disabled doCheck;
 
+  # work around python distutils compiling C++ with $CC
+  NIX_CFLAGS_COMPILE = stdenv.lib.optionalString stdenv.isDarwin "-I${libcxx}/include/c++/v1";
```

Comment:
I have hit this in a few places now (and this does indeed fix it). But, this is so weird that it only happens on Darwin. Could it be an LLVM thing? Maybe we could do this globally in LLVM's flags so that we don't have to hack around it so much.

/cc @lnl7 @ericson2314

Reply:
This only works because stdc++ and gcc are in the same package, so it knows where to look as a fallback if the stdenv doesn't add it explicitly because $CC is used. On darwin (and any other clangStdenv) libcxx and clang are separate packages, so even if clang has the same fallback it won't find anything on it's own.

But I'm not sure how this relates to https://github.com/NixOS/nixpkgs/pull/19585, I'd expect that one would be enough. (https://github.com/NixOS/nixpkgs/pull/42074#discussion_r195866881)